### ISSUE TITLE
Unhandled Promise rejection: Unable to get property 'length' of undefined or null reference ;

### ISSUE THREAD

**Issue by BogdanDumitruMarian** [service-mocker/service-mocker#45]
## Environment

- Browser: IE11
- Version of service-mocker: 2.1.1

Hi guys. There seams to be little bug in the code below:

```
self.addEventListener('message', function () {
  var _ref = _asyncToGenerator(_regenerator2.default.mark(function _callee(event) {
    var data, ports, port;
    return _regenerator2.default.wrap(function _callee$(_context) {
      while (1) {
        switch (_context.prev = _context.next) {
          case 0:
            data = event.data, ports = event.ports;
            if (!(!data || !ports.length)) {
              _context.next = 3;
              break;
            }
          ...
```

Namely, _ports_ is not always defined, hence the error in the title.

**Comment by VincentBel** [comment]
@BogdanDumitruMarian Thank you for reporting. But the [service-mocker-demo](https://service-mocker.github.io/service-mocker-demo/) works on IE11 of my virtual machine. Can you provide a concrete example?

**Comment by idiotWu** [comment]
@VincentBel `event.ports` will somehow be `null` on IE11

![image](https://user-images.githubusercontent.com/6022672/32152857-fcb5c5d6-bd69-11e7-8fd2-a5f5b8cdd1a8.png)

**Comment by dolphin-wood** [comment]
Fixed in [2.1.2](https://github.com/service-mocker/service-mocker/releases/tag/v2.1.2)

**Comment by vincentbel** [comment]
@idiotWu Oh, IE is weird. 😂

[MessageEvent.ports](https://developer.mozilla.org/en-US/docs/Web/API/MessageEvent/ports)
